.github/workflows/pre-submit.reviewdog.yml

```diff
@@ -1,5 +1,7 @@
 name: reviewdog
-on: [pull_request]
+
+"on": [pull_request_target]
+
 jobs:
   golangci-lint:
     runs-on: ubuntu-latest
```

Comment:
if you want to checkout the code at the PR, you can use:
```
with:
        ref: ${{ github.event.pull_request.head.sha }}
```


I suggest you use workflow_run + pull_request as explained in https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

I think it'd be fine to just set the permissions in the existing workflow, since we don't run the PR's code anyway. But does not hurt to follow best practices :-)

Reply:
Yeah, I'd like to be able to do that but I'm not sure the linters will work that way as the actions do the linting and PR commenting in one go. IIUC they would need to split up linting (untrusted pull_request) and PR commenting (trusted workflow_run) and the linting step would need to save intermediate linter results to an artifact.

As an aside, we should probably also consider setting the default permissions to the "restrictive" set here. I think this is the "Read repository contents permission" setting under "Actions > general"
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

Reply:
I decided to use the normal pull_request event and just not allow PR comments. We'll need to look at the logs for the check that fails instead of getting a friendly PR comment, but maybe that's ok for now.